mapie/time_series_regression.py

```diff
@@ -0,0 +1,312 @@
+            beta = np_nanquantile(
+                lower_bounds,
+                betas,
+                axis=1,
+                method="lower",
+            )  # type: ignore
+            betas_0[ind_alpha, :] = betas[
+                np.argmin(one_alpha_beta - beta, axis=0)
+            ]
+
+        return betas_0
+
+    def _pred_multi(self, X: ArrayLike) -> NDArray:
+        """
+        Return a prediction per train sample for each test sample, by
+        aggregation with matrix  ``k_``.
+
+        Parameters
+        ----------
+            X: NDArray of shape (n_samples_test, n_features)
+                Input data
+
+        Returns
+        -------
+            NDArray of shape (n_samples_test, n_samples_train)
+        """
+        y_pred_multi = np.column_stack(
+            [e.predict(X) for e in self.estimators_]
+        )
+        # At this point, y_pred_multi is of shape
+        # (n_samples_test, n_estimators_). The method
+        # ``aggregate_with_mask`` fits it to the right size
+        # thanks to the shape of k_.
+
+        y_pred_multi = self.aggregate_with_mask(y_pred_multi, self.k_)
+        return y_pred_multi
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        alpha: Optional[Union[float, Iterable[float]]] = None,
+        beta_optimize: bool = True,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        Correspond to 'Conformal prediction for dynamic time-series'.
+
+        Parameters
+        ----------
+        beta_optimize: bool
+            Whether to optimize the PIs' width or not.
+
+        """
+
+        # Checks
+        check_is_fitted(self, self.fit_attributes)
+        self._check_ensemble(ensemble)
+        alpha = cast(Optional[NDArray], check_alpha(alpha))
+        X = check_array(X, force_all_finite=False, dtype=["float64", "object"])
+        y_pred = self.single_estimator_.predict(X)
+
+        if alpha is None:
+            return np.array(y_pred)
+        else:
+            alpha_np = cast(NDArray, alpha)
+            check_alpha_and_n_samples(alpha_np, len(self.conformity_scores_))
+
+            if beta_optimize:
+                betas_0 = self._beta_optimize(
+                    alpha=alpha_np,
+                    lower_bounds=self.conformity_scores_.reshape(1, -1),
+                    upper_bounds=self.conformity_scores_.reshape(1, -1),
+                )
+            else:
+                betas_0 = np.full(
+                    shape=(len(alpha), len(self.conformity_scores_)),
+                    fill_value=np.nan,
+                    dtype=float,
+                )
+                for ind_alpha, _alpha in enumerate(alpha):
+                    betas_0[ind_alpha, :] = _alpha / 2.0
+
+            lower_quantiles = np_nanquantile(
+                self.conformity_scores_,
+                betas_0[:, 0],
+                axis=0,
+                method="lower",
+            ).T  # type: ignore
```

Comment:
Are the "type: ignore" still needed now the typing is fixed ?

Reply:
not needed